rust/ql/lib/codeql/rust/dataflow/internal/DataFlowImpl.qll

```diff
@@ -759,12 +759,17 @@ final class ReferenceContent extends Content, TReferenceContent {
 }
 
 /**
- * An element in an array.
+ * An element in a collection where we do not track the specific collection
+ * type nor the placement of the element in the collection. Therefore the
+ * collection should be one where the elements are reasonably homogeneous,
+ * i.e., if one is tainted all elements are considered tainted.
+ *
+ * Examples include the elements of a set, array, vector, or stack.
  */
-final class ArrayElementContent extends Content, TArrayElement {
-  ArrayElementContent() { this = TArrayElement() }
+final class CollectionContent extends Content, TArrayElement {
```

Comment:
`TArrayElement` should be renamed to `TElementContent` as well. This will also reveal that the MaD token `"ArrayElement"` needs to be changed in `FlowSummaryImpl.qll`. And once that is done, the QL doc in `ModelsAsData.qll` needs to be updated, as well as a few tests.